app/processing/transform-payment-request.js

```diff
@@ -1,12 +1,18 @@
-const confirmDueDates = require('./confirm-due-dates')
+const { BPS } = require('../constants/schemes')
+const { confirmPaymentRequestNumber } = require('./confirm-payment-request-number')
+const getCompletedPaymentRequests = require('./get-completed-payment-requests')
 const calculateDelta = require('./delta')
+const confirmDueDates = require('./confirm-due-dates')
 const enrichPaymentRequests = require('./enrichment')
-const getCompletedPaymentRequests = require('./get-completed-payment-requests')
 
 const transformPaymentRequest = async (paymentRequest) => {
+  // If BPS, then need to confirm payment request number as rekeyed claims can result in duplicate payment request numbers
+  if (paymentRequest.schemeId === BPS) {
+    paymentRequest.paymentRequestNumber = await confirmPaymentRequestNumber(paymentRequest)
```

Comment:
I was meaning to ask you about this anyway, but what does payment.paymentRequestNumber, deltaPaymentRequests.completedPaymentRequests etc. actually do? Where does it come from?

Reply:
`paymentRequestNumber` is the version of the payment.  So `1` = first payment, `2` is the first post payment adjustment, `3` is the second post payment adjustment etc.

In this specific line of code, BPS sometimes incorrectly outputs the wrong value, so we have a function here to check what it should be and replace it if it's wrong.

Anything starting `paymentRequst` in this context is the current payment request we are calculating and the data in it.

`completedPaymentRequests` are the final state of the request after all post payment adjustment calculations have completed and it's ready to go to D365.

Sometimes we send these to the request editor first so we attach other data to `deltaPaymentRequests` that users need to see in request editor.